flocker/control/_model.py

```diff
@@ -723,35 +757,35 @@ def move_application(self, application, target_node):
         """
         deployment = self
         for node in deployment.nodes:
-            for container in node.applications:
```

Comment:
The crux is more about being able to write a general-purpose differ and have it do  the "right thing" with our deeply pyrsistent configuration and state.

However, as a side effect there are lots of places where it was clear that what we wanted was a pmap for O(1) lookup, but instead did O(n) lookup because we had a PSet. These are just bonus points :).